manifests/user.pp

```diff
@@ -62,9 +62,17 @@
     'absent'  => 'delete',
   }
 
+  $parameters = {
+    'password' => Deferred('sqlserver::password', [$password]),
+    'database' => $database,
+    'user' => $user,
+    'login' => $login,
+    'default_schema' => $default_schema,
+  }
+
   sqlserver_tsql { "user-${instance}-${database}-${user}":
     instance => $instance,
-    command  => template("sqlserver/${create_delete}/user.sql.erb"),
+    command  => Deferred('inline_epp', [file("sqlserver/${create_delete}/user.sql.epp"), $parameters]),
```

Comment:
@binford2k added a deferrable_epp function to stdlib.

Could we use this?

https://github.com/puppetlabs/puppetlabs-stdlib/blob/b3f473f1f2e64c665d94e3ef97d00325ecd55c62/functions/deferrable_epp.pp#L8

Reply:
I have added `Deferred` looking at compatibility. The `deferrable_epp` recently got added and will not available with Puppet 7.
If we make sure it will always get used with Puppet 8 then we are good to `deferrable_epp.pp` otherwise we still need to stick with `Deferred`.
Please suggest.

Reply:
As we are pushing this change on master branch, so I think you are right we can use `deferrable_epp` over `Deferred`. Updated.

Reply:
Also I haven't see any references on doc about https://www.puppet.com/docs/puppet/8/deferring_functions.html so stick with this.

Reply:
@chelnak addressed your comment, can you please re-look. Thanks